app/src/main/java/com/team22/soundary/core/UiState.kt

```diff
@@ -0,0 +1,8 @@
+package com.team22.soundary.core
+
+sealed class UiState<out T>(val _data: T?) {
+    object Loading : UiState<Nothing>(_data = null)
+    data class Success<out R>(val data: R) : UiState<R>(_data = data)
+
+    data class Error(val message: String?) : UiState<Nothing>(_data = null)
+}
```

Comment:
p3; UiState 가 굳이 data 를 들고있지 않아도 괜찮을 것 같아요. Success 만을 위한 것이니까요..!

```suggestion
sealed class UiState<out T>{
    data object Loading : UiState<Nothing>()
    data class Success<out T>(val data: T) : UiState<T>()
    data class Error(val message: String?) : UiState<Nothing>()
}```

Reply:
변경하였습니다! 
제네릭을 처음 다루다보니 미숙했던 점이 있었던거 같습니다..!